test/array.civet

```diff
@@ -239,6 +239,58 @@ describe "array", ->
     [x=y]
   """
 
+  testCase """
+    postfix if second
+    ---
+    [
+      1
+      1 if x
+    ]
+    ---
+    [
+      1,
+      (x)?1:void 0
+    ]
+  """
+
+  testCase """
+    postfix if first
+    ---
+    [
+      1 if x
+      a
+    ]
+    ---
+    [
+      (x)?1:void 0,
+      a
+    ]
+  """
+
+  testCase """
+    postfix if with splat
+    ---
+    [
+      ...a if x
+    ]
+    ---
+    [
+      ...(x)?a:void 0
```

Comment:
This is sadly a bit broken, because `[...undefined]` doesn't work (while `{...undefined}` does...). Doesn't need to be fixed here, but would be cool. (I'm also not quite sure how to fix it...)

Reply:
One possible fix would be: if there is a `...` to fill an implicit `else` branch with a `[]`

Reply:
That would be super cool, also with non-suffix versions:

```js
[
  ...if x
    [x, y, z]
]
```

Honestly I'd kind of prefer this behavior for things like:

```js
[
  x if x?
  y if y?
]
↓↓↓
[
  ...(x != null ? [x] : []),
  ...(y != null ? [y] : [])
]
```

But I guess that might be counterintuitive / a bit too magical?  What do you think?  I've mentioned adding this kind of notation to object literals, but it's less ambiguous there:

```js
{
  x: y if y?
}
↓↓↓
{
  ...(y != null ? {x: y} : void0)
}
```